### DISCUSSION TITLE
Multi select does not respect the boundaries of the canvas

### DISCUSSION THREAD

**Discussion by varghesethomase** [xyflow/xyflow#3233] [Q&A]
I would like to understand if there is a way to limit dragging of the nodes towards the canvas boundary when a multi select is performed. 
Currently, when two widgets are selected, and is moved to the canvas boundary, it forces the farthest element from the canvas boundary to move towards it till they overlap. What is more desirable should be that when multi select is performed, the selected boundaries of the "multi selected" area is respected when moved towards the canvas boundary and retains intact how it is when the canvas boundary is touched by the selected element closest to it. I am attaching a video of how it behaves currently. I would appreciate your support on how to achieve it. 

https://github.com/wbkd/react-flow/assets/5429756/214b3e98-69b3-457a-9dfa-36fe6fc9b6c6

**Comment by moklick** [comment]
Hey @varghesethomase 

could you give us some more information? Are you using `nodeExtent` in the example?

**Comment by varghesethomase** [comment]
Yes. I am using `nodeExtent` @moklick 
```
nodeExtent={[
  [0, 0],
  [dashboardWidth, Infinity],
]}
```
also uses translateExtent
```
translateExtent={[
  [0, 0],
  [dashboardWidth, canvasHeight],
]}
```
here vertically it extends without limits, but horizontally it is only extended to a fixed width.

**Comment by chrtze** [comment]
Hey @varghesethomase, thanks for pointing this out. This is a bug and will be worked on in this issue: https://github.com/wbkd/react-flow/issues/3239

You can follow the progress on the issue for more information.

**Reply by varghesethomase** [reply]
@chrtze no problem. And thanks a lot for your support.
